plugins/catalog/src/utils/getEntitySourceLocation.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Entity,
+  parseLocationReference,
+  SOURCE_LOCATION_ANNOTATION,
+} from '@backstage/catalog-model';
+import { ConfigApi } from '@backstage/core';
+import { ScmIntegrations } from '@backstage/integration';
+
+export type EntitySourceLocation = {
+  url: string;
+  type?: string;
+};
+
+export function getEntitySourceLocation(
```

Comment:
hm so ... this isn't really for external use right? since it's in this package

just a slight confusion on location type+target vs this url+type where the type isn't a location type :) maybe call them `locationTargetUrl` and `integrationType` to make it clearer

Reply:
No, this is not exported from the catalog package, just used internally. I like the naming suggestion, updated!